extract/extract.py

```diff
@@ -1664,6 +1665,28 @@ def is_launchable(file):
 
                 request.result.add_section(new_section)
 
+    def odf_with_macros(self, request: ServiceRequest):
```

Comment:
I think this ODF heuristic/ResultSection should be moved to the oletools module, which already handles those files and extract the files that you were interested by, as well as knowing what name they have. Te actual extraction doesn't use the name from the file and instead name them as hash, so it's harder to match them, but you can test with those two malicious samples which contain StarBasic code in a file contained in a folder named "Basic". You probably want to check for case-insensitive name too.
6326ad30d56a3132df633e49280f7ed356ff9a387deeb33a1dacbb15fa4c7471
e5324991ea2715f5028e8a83bcb1f0f3d516b7a5a22e974492fa012b4889bd51

Reply:
https://github.com/CybercentreCanada/assemblyline-service-oletools/pull/143